src/main/java/com/mraof/minestuck/entity/ServerCursorEntity.java

```diff
@@ -0,0 +1,249 @@
+import software.bernie.geckolib3.core.AnimationState;
+import software.bernie.geckolib3.core.IAnimatable;
+import software.bernie.geckolib3.core.PlayState;
+import software.bernie.geckolib3.core.builder.AnimationBuilder;
+import software.bernie.geckolib3.core.builder.ILoopType;
+import software.bernie.geckolib3.core.controller.AnimationController;
+import software.bernie.geckolib3.core.event.predicate.AnimationEvent;
+import software.bernie.geckolib3.core.manager.AnimationData;
+import software.bernie.geckolib3.core.manager.AnimationFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+
+public class ServerCursorEntity extends LivingEntity implements IAnimatable, IEntityAdditionalSpawnData
+{
+	
+	private final AnimationFactory factory = new AnimationFactory(this);
+	
+	//These are only used server-side to make sure cursors are removed properly.
+	private int despawnTimer = 0;
+	private boolean removalFlag = false;
+	private final int CURSOR_DESPAWN_TICKS = 40; //The time, in ticks, that it takes for a Sburb editmode cursor to despawn after not receiving any updates.
+	private final int CURSOR_REMOVAL_PADDING = 3; //The time added, in ticks, after the removal animation has finished, but before the cursor is removed.
+	
+	// Specifies the current animation phase
+	@Nonnull
+	private Animation animation = Animation.CLICK;
+	
+	protected ServerCursorEntity(EntityType<? extends ServerCursorEntity> type, Level level)
+	{
+		super(type, level);
+		
+		noPhysics = true;
+		setNoGravity(true);
+		setInvulnerable(true);
+		
+		setBoundingBox(new AABB(0, 0, 0, 0, 0, 0));
+	}
+	
+	private <E extends IAnimatable> PlayState predicate(AnimationEvent<E> event)
+	{
+		if(!waitForFinish(event))
+			event.getController().setAnimation(new AnimationBuilder().addAnimation(animation.animationName, animation.looping ? ILoopType.EDefaultLoopTypes.LOOP : ILoopType.EDefaultLoopTypes.PLAY_ONCE));
+		
+		return PlayState.CONTINUE;
+	}
+	
+	@Override
+	public void registerControllers(AnimationData data)
+	{
+		data.addAnimationController(new AnimationController<>(this, "controller", 0, this::predicate));
+	}
+	
+	@Override
+	public AnimationFactory getFactory()
+	{
+		return this.factory;
+	}
+	
+	@Override
+	public void addAdditionalSaveData(CompoundTag compound)
+	{
+		super.addAdditionalSaveData(compound);
+		
+		compound.putInt("DespawnTimer", despawnTimer);
+		compound.putInt("Animation", animation.ordinal());
+		if(removalFlag)
+			compound.putBoolean("RemovalFlag", true);
+	}
+	
+	@Override
+	public void readAdditionalSaveData(CompoundTag compound)
+	{
+		super.readAdditionalSaveData(compound);
+		
+		if(compound.contains("DespawnTimer", Tag.TAG_ANY_NUMERIC))
+			despawnTimer = compound.getInt("DespawnTimer");
+		if(compound.contains("Animation", Tag.TAG_ANY_NUMERIC))
+			animation = Animation.values()[compound.getInt("Animation")];
+		if(compound.contains("RemovalFlag"))
+			removalFlag = compound.getBoolean("RemovalFlag");
+	}
```

Comment:
My impression is that this entity shouldn't live long enough to need serialization, so this should not be needed.

Reply:
You're probably right, yeah.